realm/realm-library/src/main/java/io/realm/log/Logger.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.log;
+
+/**
+ * Interface for custom loggers that can be registered at {@link RealmLog#add(Logger)}.
+ * The different log levels are described in {@link LogLevel}.
+ */
+public interface Logger {
+
+    /**
+     * Defines which {@link LogLevel} events this logger cares about. This value is used primarily by Realms underlying
+     * storage engine to determine which events are sent to {@link RealmLog}.
+     * <p>
+     * If multiple loggers are registered, the minimum value among all loggers is used.
+     * <p>
+     * Note that sending log events from the native layer is relatively expensive, so only set this value to events
+     * that are truly useful.
+     *
+     * @return the minimum {@link LogLevel} events this logger cares about.
+     */
+    int getMinimumLogLevel();
+
+    /**
+     * Handles a {@link LogLevel#TRACE} event.
+     *
+     * @param throwable optional exception to log.
+     * @param message optional additional message.
+     * @param args optional arguments used to format the message using {@link String#format(String, Object...)}.
+     */
+    void trace(Throwable throwable, String message, Object... args);
```

Comment:
Technically you are right, but right now each logger decided if they want to log or not. Which means that no work is done if the logger doesn't care about it. Also, it would prevent loggers from doing their own formatting if they wanted to.